ready_src/Buffer.sol

```diff
@@ -0,0 +1,110 @@
+// Copyright 2023 Cartesi Pte. Ltd.
+
+// SPDX-License-Identifier: Apache-2.0
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+pragma solidity ^0.8.0;
+
+import "./Memory.sol";
+
+library Buffer {
+    using Buffer for Buffer.Context;
+    using Memory for Memory.AlignedSize;
+
+    struct Context {
+        bytes data;
+        uint256 offset;
+    }
+
+    function consumeBytes8(Buffer.Context memory buffer)
+        internal
+        pure
+        returns (bytes8)
+    {
+        bytes8 b8 = bytes8(buffer.peekBytes32());
```

Comment:
Do you think it makes sense to create a `peekBytes8`?

Reusing the `peekBytes32` and doing a typecast makes me want to double check haha. Creating a specific bytes8 peek functions allows you to still reuse the `peekBytes32`, but have a specific test for that which will remove all doubts.

Also, this code we've never done an integration test, right? Like, does the whole replay test and other similar tests use this buffer?

I'm thinking on how we'll build this buffer offchain. If we're not doing an integration test using the buffer, maybe there's also a way to create an integration test for it like the replay. Let's think about it; maybe you can teach me how we're generating the replay tests, and we can do a similar thing here (if we're not already doing it!).

Reply:
The `replay` tests use the buffer.

Reply:
If they use it that's perfect! I thought they could be using the mocked stuff, which didn't include the buffer. Ignore the second part of my comment haha.